test/integration/credentials_test.go

```diff
@@ -0,0 +1,224 @@
+	createCredsParams := params.CreateGithubCredentialsParams{
+		Name:        dummyCredentialsName,
+		Endpoint:    defaultEndpointName,
+		Description: "GARM test credentials",
+		AuthType:    params.GithubAuthTypeApp,
+	}
+	_, err := createGithubCredentials(suite.cli, suite.authToken, createCredsParams)
+	suite.Error(err, "expected error when creating credentials with missing auth type params")
+	expectAPIStatusCode(err, 400)
+}
+
+func (suite *GarmSuite) TestGithubCredentialsUpdateFailsWhenBothPATAndAppAreSupplied() {
+	t := suite.T()
+	t.Log("Testing error when both PAT and App are supplied")
+	creds, err := suite.createDummyCredentials(dummyCredentialsName, defaultEndpointName)
+	suite.NoError(err)
+	defer suite.DeleteGithubCredential(int64(creds.ID))
+
+	privateKeyBytes, err := getTestFileContents("certs/srv-key.pem")
+	suite.NoError(err)
+	updateCredsParams := params.UpdateGithubCredentialsParams{
+		PAT: &params.GithubPAT{
+			OAuth2Token: "dummy",
+		},
+		App: &params.GithubApp{
+			AppID:           123,
+			InstallationID:  456,
+			PrivateKeyBytes: privateKeyBytes,
+		},
+	}
+	_, err = updateGithubCredentials(suite.cli, suite.authToken, int64(creds.ID), updateCredsParams)
+	suite.Error(err, "expected error when updating credentials with both PAT and App")
+	expectAPIStatusCode(err, 400)
+}
+
+func (suite *GarmSuite) TestGithubCredentialsFailWhenAppKeyIsInvalid() {
+	t := suite.T()
+	t.Log("Testing error when app key is invalid")
+	createCredsParams := params.CreateGithubCredentialsParams{
+		Name:        dummyCredentialsName,
+		Endpoint:    defaultEndpointName,
+		Description: "GARM test credentials",
+		AuthType:    params.GithubAuthTypeApp,
+		App: params.GithubApp{
+			AppID:           123,
+			InstallationID:  456,
+			PrivateKeyBytes: []byte("invalid"),
+		},
+	}
+	_, err := createGithubCredentials(suite.cli, suite.authToken, createCredsParams)
+	suite.Error(err, "expected error when creating credentials with invalid app key")
+	expectAPIStatusCode(err, 400)
+}
+
+func (suite *GarmSuite) TestGithubCredentialsFailWhenEndpointDoesntExist() {
+	t := suite.T()
+	t.Log("Testing error when endpoint doesn't exist")
+	createCredsParams := params.CreateGithubCredentialsParams{
+		Name:        dummyCredentialsName,
+		Endpoint:    "iDontExist.example.com",
+		Description: "GARM test credentials",
+		AuthType:    params.GithubAuthTypePAT,
+		PAT: params.GithubPAT{
+			OAuth2Token: "dummy",
+		},
+	}
+	_, err := createGithubCredentials(suite.cli, suite.authToken, createCredsParams)
+	suite.Error(err, "expected error when creating credentials with invalid endpoint")
+	expectAPIStatusCode(err, 404)
+}
+
+func (suite *GarmSuite) TestGithubCredentialsFailsOnDuplicateName() {
+	t := suite.T()
+	t.Log("Testing error on duplicate credentials name")
+	creds, err := suite.createDummyCredentials(dummyCredentialsName, defaultEndpointName)
+	suite.NoError(err)
+	defer suite.DeleteGithubCredential(int64(creds.ID))
```

Comment:
The golang testing package has a [Cleanup()](https://pkg.go.dev/testing#T.Cleanup) function

Reply:
Updated all suite to use `t.Cleanup` instead of `defer FunctionName`